tests/test_recipe.py

```diff
@@ -8,6 +8,13 @@
 dummy_fnames = ["a.nc", "b.nc", "c.nc"]
 
 
+def incr_date(ds, fname=""):
```

Comment:
Can you explain the point of the `fname` kwarg?

Reply:
We use `incr_date` for both `process_input` and `process_chunk` in the test, but they have a different signature:
- `process_input(ds, fname) -> ds`
- `process_chunk(ds) -> ds`

Would you like to have different hooks?

Reply:
Ok, yes I see now. Adding the fname argument in the `process_input` makes sense I guess. If we were opening files directly with xarray (rather than first with fsspec) this information would also be available in the xarray `.encoding` attributes. But that's not the case here.

Reply:
I was also confused on this point. Can we rename `fname` to `filename`? I read `fname` as function name.